communication/src/networking.rs

```diff
@@ -106,13 +127,19 @@ pub fn start_connections(addresses: Arc<Vec<String>>, my_index: usize, noisy: bo
 }
 
 /// Result contains connections [my_index + 1, addresses.len() - 1].
-pub fn await_connections(addresses: Arc<Vec<String>>, my_index: usize, noisy: bool) -> Result<Vec<Option<TcpStream>>> {
+pub fn await_connections(addresses: Arc<Vec<String>>, my_index: usize, noisy: bool, nonce: Arc<(Mutex<Option<u64>>, Condvar)>) -> Result<Vec<Option<TcpStream>>> {
     let mut results: Vec<_> = (0..(addresses.len() - my_index - 1)).map(|_| None).collect();
     let listener = TcpListener::bind(&addresses[my_index][..])?;
 
     for _ in (my_index + 1) .. addresses.len() {
         let mut stream = listener.accept()?.0;
```

Comment:
This doesn't seem right. Since some processes might crash and then reconnect multiple times, we will end up with some of the connections being `None`.